File: Framework/LanguageSupport/python/MMIPython/src/MMIPython/abstraction/access/interface/adapter_client.py
## SPDX-License-Identifier: MIT
## The content of this file has been developed in the context of the MOSIM research project.
## Original author(s): Jannes Lehwald


# -*- coding: utf-8 -*-
"""

"""

class IAdapterClient(object):
    """
    A wrapper for an adapter client connection
    
    Attributes
    ----------
    _acces : MMIAdapter.Iface
        The actual access
    """
    
    def __init__(self):
        """
        Constructor
        """
        
        self._access = None
        
    def get_adapter(self):
        """
        Returns the actual adapter
        
        Returns
        ------
        MMIAdapter.Iface
            The actual adapter
        """
        return self._access
    
    def dispose(self):
        """
        Closes the connection to the adapter.
        """
        pass

File: Framework/LanguageSupport/cs/MMICSharp/Clients/AdapterClient.cs
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Felix Gaisbauer

using MMIStandard;
using System;
using Thrift.Protocol;
using Thrift.Transport;

namespace MMICSharp.Clients
{
    /// <summary>
    /// Basic client for an adapter
    /// </summary>
    public class AdapterClient:ClientBase
    {

        /// <summary>
        /// The access of the client
        /// </summary>
        public MMIAdapter.Client Access;

        /// <summary>
        /// Basic constructor
        /// </summary>
        /// <param name="address"></param>
        /// <param name="port"></param>
        public AdapterClient(string address, int port, bool autoStart = true):base(address,port,autoStart)
        {
        }


        protected override void AssignAccess(TProtocol protocol)
        {
            this.Access = new MMIAdapter.Client(protocol);
        }
    }
}
